product/tests/test_product.py

```diff
@@ -28,6 +28,12 @@ def test_save_product_and_delete_product(self, saved_product0):
         assert saved_product0 in Product.objects.all()
         Product.delete_product(saved_product0.qr_code)
         assert saved_product0 not in Product.objects.all()
+        res = True
+        try:
+            Product.delete_product("NOTEXISTEDTEST")
+        except Product.DoesNotExist:
+            res = False
+        assert not res
```

Comment:
We don't typically do try-except inside tests. Instead, we assert that the exception was raised with context manager `with pytest.raises(SomeException) as e`.

See examples here: https://stackoverflow.com/a/23514853

Reply:
Fixed